GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Gherkin.Ast;
+using GherkinSyncTool.Models;
+using GherkinSyncTool.Models.Configuration;
+using GherkinSyncTool.Synchronizers.AllureTestOps.Model;
+using Quantori.AllureTestOpsClient.Model;
+using Scenario = Gherkin.Ast.Scenario;
+using Step = Gherkin.Ast.Step;
+
+namespace GherkinSyncTool.Synchronizers.AllureTestOps.Content;
+
+public class CaseContentBuilder
+{
+    private readonly AllureTestOpsSettings _allureTestOpsSettings = ConfigurationManager.GetConfiguration<AllureTestOpsConfigs>().AllureTestOpsSettings;
+    
+    public CreateTestCaseRequest BuildCaseRequest(Scenario scenario, IFeatureFile featureFile)
+    {
+        var steps = GetSteps(scenario, featureFile);
+
+        var caseRequest = new CreateTestCaseRequest
+        {
+            Name = scenario.Name,
+            ProjectId = _allureTestOpsSettings.ProjectId,
+        };
+
+        return caseRequest;
+    }
+    
+    private List<string> GetSteps(Scenario scenario, IFeatureFile featureFile)
+    {
+        var scenarioSteps = ExtractSteps(scenario.Steps.ToList());
+
+        var background = featureFile.Document.Feature.Children.OfType<Background>().FirstOrDefault();
+        if (background is not null)
+        {
+            var backgroundSteps = ExtractSteps(background.Steps.ToList());
+            return backgroundSteps.Concat(scenarioSteps).ToList();
+        }
+
+        return scenarioSteps;
+    }
+
+    private List<string> ExtractSteps(List<Step> steps)
+    {
+        List<string> resultSteps = new List<string>();
+        foreach (var step in steps)
+        {
+            var keywordFormatted = $"**{step.Keyword.Trim()}** ";
+            var stepFormatted = step.Text.Replace("<", "***").Replace(">", "***");
+
+            var fullStep = keywordFormatted + stepFormatted;
+
+            if (step.Argument is DocString docString)
+            {
+                fullStep += Environment.NewLine + docString.Content;
+            }
+
+            if (step.Argument is DataTable table)
+            {
```

Comment:
How many steps could be there approximately? Maybe using of StringBuilder will be better here